presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java

```diff
@@ -0,0 +1,157 @@
+     * @param threadInfo the thread information
+     * @param stackTracePos the stackTrace position
+     * @return string builder with ThreadState information on Blocks/Waits
+     */
+    private static StringBuilder appendThreadStateInfo(StringBuilder sb, ThreadInfo threadInfo, int stackTracePos)
+    {
+        if (stackTracePos != 0 || threadInfo == null) {
+            return sb;
+        }
+
+        Thread.State ts = threadInfo.getThreadState();
+        switch (ts) {
+            case BLOCKED:
+                sb.append("\t-  BLOCKED on " + threadInfo.getLockInfo());
+                sb.append("\n\t- BLOCKED for " + Duration.ofNanos(threadInfo.getBlockedTime()).toSeconds() + "s");
+                break;
+            case WAITING:
+            case TIMED_WAITING:
+                sb.append("\t-  WAITING on " + threadInfo.getLockInfo());
+                sb.append("\n\t- WAITED for " + Duration.ofNanos(threadInfo.getWaitedTime()).toSeconds() + "s");
+                break;
+            default:
+        }
+
+        return sb;
+    }
+
+    /**
+     * Appends the Locked Synchronizers, if any, to the log string
+     * @param sb the string builder of the log string
+     * @param locks the lock information from the thread info
+     * @return the log message with appended locked synchronizer list, (class name/hashcode)
+     */
+    private static StringBuilder appendLockedSynchronizers(StringBuilder sb, LockInfo[] locks)
+    {
+        if (locks.length <= 0) {
+            return sb.append('\n');
+        }
+
+        sb.append("\n\tNumber of LOCKED Synchronizers: " + locks.length);
+        for (LockInfo li : locks) {
+            sb.append("\n\t\tClass Name: " + li.getClassName());
+            sb.append("\n\t\tHash Code: " + li.getIdentityHashCode());
+            sb.append('\n');
+        }
+
+        return sb.append('\n');
+    }
+
+    /**
+     * Appends the locked monitor list to the log message
+     * @param sb the string builder of the log string
+     * @param miList the list of locked monitors
+     * @param stackTracePos the stack trace position
+     * @return the log message with appended locked monitor list
+     */
+    private static StringBuilder appendLockedMonitors(StringBuilder sb, MonitorInfo[] miList, int stackTracePos)
+    {
+        if (miList.length <= 0) {
+            return sb;
+        }
+
+        sb.append("\n\tLOCKED Monitors\n\t");
+        for (MonitorInfo mi : miList) {
+            // skip if we are not on the locked thread
+            if (stackTracePos != mi.getLockedStackDepth()) {
+                continue;
+            }
+
+            sb.append("\t- " + mi);
+            sb.append('\n');
+        }
+
+        return sb;
+    }
+
+    /**
+     * Returns the full stacktrace string of the thread
+     * @param threadInfo the thread information
+     * @return the full stacktrace of the thread information
+     */
+    public static String getFullStackTrace(ThreadInfo threadInfo)
```

Comment:
Please put public at the top, let's order method by usage. So first public, then methods used in public method, then methods that were used by methods above...

Reply:
Gotchya,  I can push these changes soon.